データベース設計

hiyak423 opened the pull request:
# what
データベースのER図作成、READMEに設計記述
# why
データベース構築のため
# ER図 画像
https://gyazo.com/9fd187af317311a01e6ced9953f966ff

hiyak423 committed:
ER図作成

hiyak423 committed:
データベース設計記述

yuri240 commented on README.md:
```diff
@@ -22,3 +22,47 @@ Things you may want to cover:
+
+## table_name
+Column          Type    Options
+
```
以下のように全体的なレイアウトを整えましょう！

```
## users テーブル

| Column             | Type                | Options                 |
|--------------------|---------------------|-------------------------|
| first_name         | string              | null: false             |
| last_name          | string              | null: false             |
| first_name_kana    | string              | null: false             |
| last_name_kana     | string              | null: false             |
| birth_date         | date                | null: false             |

```

yuri240 commented on README.md:
```diff
@@ -22,3 +22,47 @@ Things you may want to cover:
+last_name_kana  string  null:false
+birth_year      integer null:false
+birth_month     integer null:false
+birth_day       integer null:false
```
```
birth_year      integer null:false
birth_month     integer null:false
birth_day       integer null:false
```
誕生日のカラムは「date型」を用いることで年、月、日を1つにして保存できます。


現状だと３つカラムを設けていただいておりますが、
１つのカラムに変更し、date型で定義しましょう！

yuri240 commented on README.md:
```diff
@@ -22,3 +22,47 @@ Things you may want to cover:
+status         integer  null:false
+shipping_cost  integer  null:false
+shipping_area  integer  null:false
+shipping_day   integer  null:false
```
**商品の状態, 配送料の負担, 発送元の地域,  発送までの日数, カテゴリー**のカラムは
active_hashで実装の予定のため、カラム名には`「_id」`をつけ、`integer`型にしましょう！

詳しくは「`最終課題補足カリキュラム 〜カテゴリーの選択を工夫しよう〜`」のカリキュラムを参考にしていただけると幸いです。

yuri240 commented on README.md:
```diff
@@ -22,3 +22,47 @@ Things you may want to cover:
+shipping_area  integer  null:false
+shipping_day   integer  null:false
+price          integer  null:false
+user_id        references null:false foreign_key: true
```
`user_id        references null:false foreign_key: true`

references型の場合、「_id」は省略可能です。
そのためカラム名はuserにしましょう。

yuri240 commented on README.md:
```diff
@@ -22,3 +22,47 @@ Things you may want to cover:
+
+## orders
+user_id        references null:false foreign_key: true
+item_id        references null:false foreign_key: true
```
54,55,66行目のカラムも`_id`を省略しましょう！

yuri240 commented on README.md:
```diff
@@ -22,3 +22,47 @@ Things you may want to cover:
+
+## shippings
+postcode       string   null:false
+prefecture     integer  null:false
```
```
prefecture     integer  null:false
```
こちらの都道府県カラムは、
active_hashのデータとして使い回すことができますので、
itemsテーブルの都道府県カラムと同じものを使用しましょう。

yuri240 commented on README.md:
```diff
@@ -22,3 +22,47 @@ Things you may want to cover:
+user_id        references null:false foreign_key: true
+item_id        references null:false foreign_key: true
+### Association
+has_one order, belongs_to user
```
ordersテーブルには、`item`と`shipping`のアソシエーションを設定しましょう！

yuri240 reviewed (commented):
レビュー依頼ありがとうございます！
コメントさせていただきましたので、ご確認お願いいたします。

また、次回ご提出の際にはER図も修正が必要となります。

お手数ではありますが、
修正後のER図を再度こちらのプルリクエストに添付していただくようお願いいたします。

hiyak423 committed:
ER図、README修正

hiyak423 committed:
Revert "ER図、README修正"

This reverts commit acf8f232ad2e223464b6fe9a08831a8bfe016689.

hiyak423 committed:
ER図、README修正(2回目)

hiyak423 commented:
yuri240さん ご確認いただきありがとうございました。

ER図ととREADMEの記述を修正しました。
ER図は、下記画像からご確認をお願いいたします。

https://gyazo.com/e9999257949969fd975d8db5286a3fb9

yuri240 commented on README.md:
```diff
@@ -22,3 +22,51 @@ Things you may want to cover:
 
 * ...
+
+## users テーブル
```
パスワードに関するカラムが不足しているようです。
適切なカラム名で１つ用意しましょう！

yuri240 commented on README.md:
```diff
@@ -22,3 +22,51 @@ Things you may want to cover:
+| last_name_kana     | string              | null:false                   |
+| birthday           | data                | null:false                   |
+### Association
+  has_many items, has_many orders
```
アソシエーションは以下のように記述しましょう！

```
 has_many :items
 has_many :orders
```

yuri240 commented on README.md:
```diff
@@ -22,3 +22,51 @@ Things you may want to cover:
+| last_name          | string              | null:false                   |
+| first_name_kana    | string              | null:false                   |
+| last_name_kana     | string              | null:false                   |
+| birthday           | data                | null:false                   |
```
`| birthday           | data                | null:false                   |`

birthdayカラムはdate型で設計しましょう！※スペルに誤りがあるようです。

yuri240 commented on README.md:
```diff
@@ -22,3 +22,51 @@ Things you may want to cover:
+| price              | integer             | null:false                   |
+| user               | references          | null:false foreign_key: true |
+### Association
+  has_one order, belongs_to user 
```
同様に51,59,72行目のアソシエーションも適切な記述に修正しましょう！

yuri240 reviewed (commented):
レビュー依頼ありがとうございます！
キャプチャ画像と修正点確認させていただきました。

追加でコメントさせていただきましたので、ご確認お願いします。

あと一息かと思いますので頑張りましょう！

hiyak423 committed:
ER図、README修正(3回目)

ER図修正
password >> encrypted_password

README修正
has_manyの記述修正
userカラムに2つ追加
>> encrypted_password
>> email

hiyak423 commented:
ご確認ありがとうございます。
ご指摘いただいた箇所を修正しました。ご確認をお願いいたします。

yuri240 commented on README.md:
```diff
@@ -22,3 +22,57 @@ Things you may want to cover:
+| first_name_kana    | string              | null:false                   |
+| last_name_kana     | string              | null:false                   |
+| birthday           | date                | null:false                   |
+| encrypted_password | string              | null:false,unique: true      |
```
`| encrypted_password | string              | null:false,unique: true      |
`

encrypted_passwordカラムからユニーク制約を外しましょう。

■理由
要件においてencrypted_passwordカラムにユニーク制約が求められていないためになります。

hiyak423 replied:
ありがとうございます。修正し、コミットしました。
ご確認をお願いいたします。

yuri240 reviewed (commented):
レビュー依頼ありがとうございます！
修正点確認させていただきました。

追加でコメントさせていただきましたので、ご確認お願いします。

hiyak423 committed:
README修正

users テーブル
passwordのunique: true削除

yuri240 commented:
早速の修正とレビュー依頼ありがとうございます。

しっかりと設計できていることが確認できましたので、
**LGTM**とさせていただきます！

こちらのブランチは忘れずにMergeを行っていただき、
ローカルのMasterにPullしておきましょう。

データベース設計の実装、大変お疲れ様でした。
引き続き、応援しております。
